src/Entity/Membership.php

```diff
@@ -200,6 +200,11 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       ->setDisplayConfigurable('view', TRUE)
       ->setRevisionable(TRUE)
       ->setSetting('workflow_callback', ['\Drupal\membership\Entity\Membership', 'getWorkflowId']);
+    $fields['provider'] = BaseFieldDefinition::create('membership_provider_id')
```

Comment:
Not sold that the configuration for a provider belongs on the Membership class -- still thinking this goes on the MembershipType.